lib/src/widget/bidirectional_paging_list_view.dart

```diff
@@ -474,9 +474,9 @@ class _BidirectionalPagingListViewState<Key, Value>
       // Check if the index corresponds to near the top or bottom based on the
       // 'reverse' flag.
       final nearTop =
-          reverse ? index == itemCount - fetchIndex : index == fetchIndex;
+          reverse ? index >= itemCount - fetchIndex : index <= fetchIndex;
       final nearBottom =
-          reverse ? index == fetchIndex : index == itemCount - fetchIndex;
+          reverse ? index <= fetchIndex : index >= itemCount - fetchIndex;
```

Comment:
Hello, thank you for reply! It seems that checking exact item index with == will still not work even for ```min(fetchIndex, length)```. There is an example that will break it:

1. Suppose id of message is the key of page (e.g. start page from message with id = ?)
2. Initial key = 20 (message id), page size = 45, prefetchIndex = 22
3. We cannot use == with min(fetchIndex, len) since we have not reached item with index 22 (because we are loaded page from item=20, but it has initial index = 0, meaning that we have to scroll down to item 20+22=42 to trigger prepend operation)

Solution seems to be straightforward: in PageFetcher we disallow to trigger multiple _doLoad calls (we cannot do it on Ui, because of ```WidgetsBinding.instance.addPostFrameCallback { pager.load(...) }```), which will protect pager from multiple requests of items that have passed fetchIndex

Reply:
Okay I understood your issue. I think we can add an additional check for the first check.

If(pages.legth == 1) i.e only the first page has been loaded, the prefetchCount is set to 0.

I think it should fix your issue right?